File: scala-problems/src/main/scala/list/RemoveDuplicatesFromSortedList.scala
package org.whsv26.leetcode
package list

object RemoveDuplicatesFromSortedList extends App {
  object Solution {
    def deleteDuplicates(head: ListNode): ListNode = {
      var node = head

      while (node != null && node.next != null) {
        if (node.x == node.next.x) {
          node.next = node.next.next
        } else {
          node = node.next
        }
      }

      head
    }
  }
}

File: golang-problems/remove_duplicates_from_sorted_list.go
package golang_problems

func deleteDuplicates(head *ListNode) *ListNode {
	node := head

	for node != nil && node.Next != nil {
		if node.Val == node.Next.Val {
			node.Next = node.Next.Next
		} else {
			node = node.Next
		}
	}

	return head
}
